### PR TITLE
feat_grid_activity:REST type changes and filterEnable

### PR DESCRIPTION
**Pull request by kapilchothe1**
Added REST type changes like pagination,sorting,filtering, searching and also added changes for filterEnablefields. 
Added Activity navigation menu and its functionality.

### CODE REVIEW

File: config/apps/event/rest/index.js [hunk 1]
```diff
@@ -1,14 +1,48 @@
...
-    // 2. pagination
-    // 3. search
-    // 4. filters
-    const res = await this.$axios.$get(
-      `https://event.test.bitpod.io/svc/api/${modelName}s?filter={"limit":"10"}`
+
+    // pagination
+    const page = options.page || 1
+    const itemsPerPage = options.itemsPerPage || 10
+    const skip = (page - 1) * (itemsPerPage || 10)
```
Comment (SukantGujar) [inline config/apps/event/rest/index.js:10]: Don't need || 10 here

File: config/apps/event/rest/index.js [hunk 2]
```diff
@@ -1,14 +1,48 @@
...
+      filters: options.filters,
+      filter: this.filter,
+      contentName,
+      ctx: this,
+    })
+
+    const filter = { limit, skip, order, where }
+
+    const resPromise = this.$axios.$get(
+      `https://event.test.bitpod.io/svc/api/${modelName}s?filter=${JSON.stringify(
```
Comment (SukantGujar) [inline config/apps/event/rest/index.js:32]: The url hostname should come from env variables or from the Nuxt router url.

**Review by SukantGujar — CHANGES_REQUESTED** [review]

File: config/apps/event/rest/index.js (hunk 1 above)
Comment (kapilchothe1) [reply config/apps/event/rest/index.js:10]: When removed '|| 10' from line 9 , page is getting hang for sometime when page loads first time due to large data.

File: config/apps/event/rest/index.js (hunk 2 above)
Comment (kapilchothe1) [reply config/apps/event/rest/index.js]: Resolved

**Review by SukantGujar — APPROVED** [review]

**Pull request merged**